src/main/java/org/jetbrains/bsp/bazel/server/bsp/BazelBspServerBuildManager.java

```diff
@@ -0,0 +1,365 @@
+                target.setDataKind(BuildTargetDataKind.SCALA);
+                target.setTags(Lists.newArrayList(ParsingUtils.getRuleType(rule.getRuleClass())));
+                target.setData(buildTarget);
+              });
+    } else if (extensions.contains(Constants.JAVA) || extensions.contains(Constants.KOTLIN)) {
+      target.setDataKind(BuildTargetDataKind.JVM);
+      target.setTags(Lists.newArrayList(ParsingUtils.getRuleType(rule.getRuleClass())));
+      target.setData(getJVMBuildTarget());
+    }
+    return target;
+  }
+
+  private List<BuildTarget> getBuildTargetForProjectPath(String projectPath) {
+    Build.QueryResult queryResult =
+        queryResolver.getQuery(
+            "query",
+            "--output=proto",
+            "--nohost_deps",
+            "--noimplicit_deps",
+            String.format(
+                "kind(binary, %s:all) union kind(library, %s:all) union kind(test, %s:all)",
+                projectPath, projectPath, projectPath));
+    return queryResult.getTargetList().stream()
+        .map(Build.Target::getRule)
+        .filter(rule -> !rule.getRuleClass().equals("filegroup"))
+        .map(this::getBuildTargetForRule)
+        .collect(Collectors.toList());
+  }
+
+  private Optional<ScalaBuildTarget> getScalaBuildTarget() {
+    if (scalacClasspath == null) {
+      buildTargetsWithBep(
+          Lists.newArrayList(
+              new BuildTargetIdentifier(
+                  "@io_bazel_rules_scala_scala_library//:io_bazel_rules_scala_scala_library"),
+              new BuildTargetIdentifier(
+                  "@io_bazel_rules_scala_scala_reflect//:io_bazel_rules_scala_scala_reflect"),
+              new BuildTargetIdentifier(
+                  "@io_bazel_rules_scala_scala_compiler//:io_bazel_rules_scala_scala_compiler")),
+          Lists.newArrayList(
+              "--aspects=@//.bazelbsp:aspects.bzl%scala_compiler_classpath_aspect",
+              "--output_groups=scala_compiler_classpath_files"));
+      List<String> classpath =
+          bepServer.getCompilerClasspath().stream().map(Uri::toString).collect(Collectors.toList());
+      List<String> scalaVersions =
+          classpath.stream()
+              .filter(uri -> uri.contains("scala-library"))
+              .collect(Collectors.toList());
+      if (scalaVersions.size() != 1) {
+        return Optional.empty();
+      }
+      String scalaVersion =
+          scalaVersions
+              .get(0)
+              .substring(
+                  scalaVersions.get(0).indexOf("scala-library-") + 14,
+                  scalaVersions.get(0).indexOf(".jar"));
+      scalacClasspath =
+          new ScalaBuildTarget(
+              "org.scala-lang",
+              scalaVersion,
+              scalaVersion.substring(0, scalaVersion.lastIndexOf(".")),
+              ScalaPlatform.JVM,
+              classpath);
+      scalacClasspath.setJvmBuildTarget(getJVMBuildTarget());
+    }
+
+    return Optional.of(scalacClasspath);
+  }
+
```

Comment:
What do you think of this method living in a scala specific file?

Reply:
You mean putting it somewhere in _ScalaBuildServerService_ or extracting one from this file?

Reply:
`ScalaBuildServerService` sounds like a good place for this to live. Wdyt?

Reply:
I've tried it now but it's a bit tricky because this method (``getScalaBuildTarget``) is called from ``getBuildTargetForRule`` and is bound to some other things here. And we can't have a dependency ``BazelBspServerBuildManager -> Service`` because it would be cyclical and against the idea of those services (the service classes should only be kept in the _Impl_ classes and delegate the work into components like ``BuildManager`` or ``RequestHandlers`` and not the other way around). Maybe this will also be easier to do after [this one](https://github.com/JetBrains/bazel-bsp/pull/24#discussion_r549333457), I'll mention it in the issue.

Reply:
Sounds great. Let's look into this then.